eds/plugin.py

```diff
@@ -70,3 +71,8 @@ def descendants(self) -> List[Plugin]:
             plugins += plugin.children
             plugins.append(plugin)
         return plugins
+
+
+PluginInterface.register(Plugin)
```

Comment:
since `Plugin` is going to inherit from `PluginInterface`, we don't need to `register`

Reply:
I was curious whether the abstract method requirement would be true with registration alone (without inheritance), but it doesn't work.  So, it means to get the validation we want, it will have to work via inheritance.  Given that, I don't see any need to require or mention that plugin writers use `register`.

Reply:
I am removing all `register` calls